metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/GrokParser.java

```diff
@@ -150,8 +149,8 @@ public void init() {
 
       if (message.size() == 0)
         throw new RuntimeException("Grok statement produced a null message. Original message was: "
-                + originalMessage + " and the parsed message was: " + message + " . Check the pattern at: "
-                + grokPath);
+                + originalMessage + " , parsed message was: " + message + " , pattern was: "
+                + grokPattern);
```

Comment:
Instead of "+ grokPattern", would you consider:
+ (LOG.isDebugEnabled() ? grokPattern : (patternLabel + " (Turn on DEBUG logging to see pattern text.)"))

But if you think it's really important to log the full pattern regardless of DEBUG setting, I can accept that.  It's just that under load in the field you might run 100,000's of messages through a faulty grokPattern before realizing the error, and blow out your log storage.  Which irritates users :-)